### ISSUE TITLE
Moving average vs. CIC filter design, Terminology

### ISSUE THREAD

**Issue by dsvf** [chipmuenk/pyFDA#152]
Two points to consider when adding a CIC mode to pyfda:

1. When designing a moving average filter which averages over 10 samples, then the number of delays is obviously only 9 since I can use the latest, non-delayed sample. Setting M to 10 returns an 11-tap FIR filter.
When designing a CIC filter, to achieve decimation-by-10, the user would have to set M to 9 to obtain the correct response. It might be advisable to either notify the user about this, use a different name for the input parameter or increment it (internally/silently?)

1. From what I can find, in CIC filters the parameter "M" denotes the number of delays inside one integrator/differentiator and in most cases is set to 1. The parameter "N" is the number of integrator and differentiator stages. This collides in terminology with the MA filter UI, where "M" is the decimation ratio (off-by-one), and "Stages" is the number of stages (alright).

**Comment by chipmuenk** [comment]
Thanks for your feedback! I agree that my naming conventions are a bit questionable. For MA filters I decided to keep parameter name _N_ for the _total_ order (as in e.g. number of zeros) obtained by cascading identical stages of simple MA filters with _M_ delays each (not the same as the number of delays inside CIC stages, AFAIK this is either 1 or 2).

Concerning CIC filter, it was probably a bad idea to confuse the user in #149 by saying "well, a MA filter is more or less the same as a CIC filter". Well, it isn't, so it would be better to design a new filter class (that also allows decimation / interpolation) with the parameters R or I (decimation / interpolation ratio resp. order of the comb filter), _M_ (1 or 2, number of delays in the CIC stages) and "stages" (how many times are integrators / comb filters cascaded). What do you think?

[CIC-MA_filter.pdf](https://github.com/chipmuenk/pyFDA/files/3426866/CIC-MA_filter.pdf)

[CIC-simple.pdf](https://github.com/chipmuenk/pyFDA/files/3426875/CIC-simple.pdf)

**Comment by dsvf** [comment]
I'd like to try some ideas out myself, but I am unable to get pyfda(x) to run from the source. Do you have a short hint on how you start the program itself from the source files?

**Comment by chipmuenk** [comment]
Did you receive my mail (about a week ago)? Were you able to start pyfda?

**Comment by dsvf** [comment]
No, I did not receive any email. I have now added it to my github profile.

**Comment by chipmuenk** [comment]
Ah, I see. I've added the info that I tried to mail you to the github project Readme (where it belongs, I guess). Please give it another try and tell me whether you had success.

**Comment by dsvf** [comment]
Thanks! I had an unrelated problem with multiple python versions interfering with each other, but now it works.
